packages/shared_preferences/shared_preferences_tool/lib/main.dart

```diff
@@ -17,10 +17,22 @@ class _SharedPreferencesTool extends StatelessWidget {
 
   @override
   Widget build(BuildContext context) {
-    return const DevToolsExtension(
-      child: SharedPreferencesStateProvider(
-        child: SharedPreferencesBody(),
-      ),
+    return DevToolsExtension(
+      child: Builder(builder: (BuildContext context) {
+        return FutureBuilder<Object>(
+            future: serviceManager.onServiceAvailable,
+            builder: (BuildContext context, AsyncSnapshot<Object> snapshot) {
+              if (snapshot.connectionState == ConnectionState.waiting) {
```

Comment:
@kenzieschmoll with the library update I had to perform this check, without this `serviceManager.service!` crashes.

Reply:
You are likely hitting a race condition where the service manager initialization has not yet completed. If `SharedPreferencesBody` requires the vm service to be initialized then this solution seems reasonable. Another would be to show a circular progress indicator because I assume this "Please connect to a VM service" message is not on the screen for long as is. In other DevTools screens, we often call `await serviceConnection.serviceManager.onServiceAvailable;` in controller initialization before interacting with the VM service.

Reply:
Got it, I'd need to change my dependencies if I do like this. Maybe the progress indicador is the better option. I'll use a future builder with `serviceConnection.serviceManager.onServiceAvailable`, this should solve the problem and keep the current architecture.